### ISSUE TITLE
Repetitive quests, such as building types

### ISSUE THREAD

**Issue by Phyks** [westnordost/StreetComplete#1190]
Hi,

**Use case**

I recently noticed that there was a (new?) quest type enabled by default "building type" (and related number of floors quest). This results in a lot of such quests to fill in, in urban areas. Even in Paris, where I live and would expect a lot of OSM contributors to have filled in such tags for buildings, this results in virtually selecting building type for every building.

I know this can be easily disabled from the settings but for a newcomer launching StreetComplete for the first time, this might not be the best first impression to have a huge number of similar quest at first launch. I would expect (but might be misleading) that the best first impression for a newcomer would be with some diversity in the quests, to have a general overview. Same for long run contributors which might be reluctant to fill in always the same quest. This tends to become a repetitive task and lose a bit from the gamification side.

I noticed this in a few places in France, probably due to the way the buildings were initially imported, without type or use.

**Proposed Solution**

Not sure what the best solution here would be. Here are a few ideas:

* The easiest solution could be to simply disable "building type" quest in France. This has the downside of having a lot less people filling in this quest then.
* A slightly more elaborated (and generic) solution could be to disable a quest if there is a too high density of this quest around the user. This has a similar downside in areas where the quest is most needed.
* The most elaborate approach could be to simply randomize the quests priority. This is not always possible as there are some implications between the quests. Typically for restaurants asking for opening hours and compliance with diets, this could be 50% of the time opening hours first and 50% of the time compliance with diets first. Same thing could be done with building types.

Hope this feedback can be useful :)
Thanks for all the work on StreetComplete!

**Comment by ENT8R** [comment]
> The easiest solution could be to simply disable "building type" quest in France. This has the downside of having a lot less people filling in this quest then.

Wait, what? What is the sense of this quest then if we simply disable it, because it would be too much?

> The most elaborate approach could be to simply randomize the quests priority.

Related to https://github.com/westnordost/StreetComplete/issues/1091#issuecomment-393640424

**Comment by rugk** [comment]
BTW, have a look [at this FAQ entry](https://wiki.openstreetmap.org/wiki/StreetComplete/FAQ#Why_does_StreetComplete_show_so_few_quests_when_I_first_start_the_app.3F).

**Comment by Phyks** [comment]
> Wait, what? What is the sense of this quest then if we simply disable it, because it would be too much?

Yes, that's why I'm not a fan at all of this solution. I was just not sure whether this was the intended behavior here to have quests with so many things to answer, or whether SC rather aimed at completing stuff, meaning completing tags which were already widespread in a given area. :)

**Comment by westnordost** [comment]
This issue will be somewhat mitigated by that starting with the next version, the maximum number of quests of one type that are downloaded in an area will be limited (to 2000).

I implemented this because of massive performance problems. In Hamburg, where I live, a download yielded more than 24000 building type quests, which ... took a while to insert into the local database.

This change will also have the side effect that the quests displayed will be more "diverse" because not one quest type can dominate the map this much. Good to know that there is a reason beside performance to think about limiting the number of quests per type per area even more.

I will close this issue now because I consider it fixed by the imposed download limit, however the limit of 2000 was just something I came up with because this takes about 12s to insert to DB, which is the upper limit I consider as being okay. It could be lowered further for usability reasons like this, after the app is tested with that lower limit.
